src/catch2/reporters/catch_reporter_listening.cpp

```diff
@@ -15,151 +17,157 @@ namespace Catch {
             listener->benchmarkFailed(error);
         }
-        m_reporter->benchmarkFailed(error);
     }
 
     void ListeningReporter::testRunStarting( TestRunInfo const& testRunInfo ) {
         for ( auto const& listener : m_listeners ) {
             listener->testRunStarting( testRunInfo );
         }
-        m_reporter->testRunStarting( testRunInfo );
     }
 
     void ListeningReporter::testGroupStarting( GroupInfo const& groupInfo ) {
         for ( auto const& listener : m_listeners ) {
             listener->testGroupStarting( groupInfo );
         }
-        m_reporter->testGroupStarting( groupInfo );
     }
 
 
     void ListeningReporter::testCaseStarting( TestCaseInfo const& testInfo ) {
         for ( auto const& listener : m_listeners ) {
             listener->testCaseStarting( testInfo );
         }
-        m_reporter->testCaseStarting( testInfo );
     }
 
     void ListeningReporter::sectionStarting( SectionInfo const& sectionInfo ) {
         for ( auto const& listener : m_listeners ) {
             listener->sectionStarting( sectionInfo );
         }
-        m_reporter->sectionStarting( sectionInfo );
     }
 
     void ListeningReporter::assertionStarting( AssertionInfo const& assertionInfo ) {
         for ( auto const& listener : m_listeners ) {
             listener->assertionStarting( assertionInfo );
         }
-        m_reporter->assertionStarting( assertionInfo );
     }
 
     // The return value indicates if the messages buffer should be cleared:
     bool ListeningReporter::assertionEnded( AssertionStats const& assertionStats ) {
-        for( auto const& listener : m_listeners ) {
-            static_cast<void>( listener->assertionEnded( assertionStats ) );
+        const bool alwaysReport =
+            assertionStats.assertionResult.getResultType() != ResultWas::Ok ||
+            m_config->includeSuccessfulResults();
+
+        for ( auto const& listener : m_listeners ) {
+            if ( alwaysReport ||
+                 listener->getPreferences().shouldReportAllAssertions ) {
+                static_cast<void>( listener->assertionEnded( assertionStats ) );
+            }
         }
-        return m_reporter->assertionEnded( assertionStats );
+        // the return value is unused, so it does not really matter what we
+        // return
+        return true;
     }
 
     void ListeningReporter::sectionEnded( SectionStats const& sectionStats ) {
         for ( auto const& listener : m_listeners ) {
             listener->sectionEnded( sectionStats );
         }
-        m_reporter->sectionEnded( sectionStats );
     }
 
     void ListeningReporter::testCaseEnded( TestCaseStats const& testCaseStats ) {
+        if ( m_preferences.shouldRedirectStdOut && m_haveNoncapturingListeners ) {
+            if ( !testCaseStats.stdOut.empty() ) {
+                Catch::cout() << testCaseStats.stdOut << std::flush;
+            }
+            if ( !testCaseStats.stdOut.empty() ) {
```

Comment:
```suggestion
            if ( !testCaseStats.stdErr.empty() ) {
```

I didn't check the actual name, but...